### ISSUE TITLE
Fix Request: HttpInterceptorNotReleasingOnException not supporting try-with-resources

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#633]
```
class ValidatingClientHttpRequestInterceptorTryResources implements ClientHttpRequestInterceptor {
    public ClientHttpResponse intercept(HttpRequest request, byte[] body, ClientHttpRequestExecution execution) throws IOException {
        try (ClientHttpResponse response = execution.execute(request, body)) {
            if (!response.getHeaders().containsKey("X-Some-Header")) {
                // Log error
                throw new MyException("validation error"); // <-- false positive
            }
            return response;
        }
    }
}
```

**Comment by jborgers** [comment]
In the above 'solution' the response is always closed. This means later executing interceptors will hit a closed response and I wonder what the response to the user will be when it is closed already. 
So although it seems nice and proper, it is not a valid solution for how the framework works.
